### PR TITLE
Color docs

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
Based on discussion in gitter.

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -10,12 +10,22 @@ Color _scaleAlpha(Color a, double factor) {
...
+  /// * Bits 8-15 are the green value.
+  /// * Bits 0-7 are the blue value.
+  ///
+  /// In other words, if AA is the alpha value in hex, RR the red value in hex,
+  /// GG the green value in hex, and BB the blue value in hex, a color can be
+  /// expressed as `const Color(0xAARRGGBB)`.
+  ///
+  /// For example, to get a fully opaque orange, you would use `const
+  /// Color(0xFFFF9000)` (`FF` for the alpha, `FF` for the red, `90` for the
+  /// green, and `00` for the blue).
```
Comment (abarth) [inline lib/ui/painting.dart]: You might want to give a specific example of translating `#FF9000` into this format as the `#` form of naming colors is very widely used (e.g., in CSS as well as Photoshop).

**Comment by abarth** [comment]
LGTM

File: lib/ui/painting.dart (hunk 1 above)
Comment (Hixie) [inline lib/ui/painting.dart]: Ok I added some prose to the class-level docs.

**Comment by abarth** [comment]
Gold star

**Comment by xster** [comment]
re: https://github.com/flutter/engine/pull/3310#issuecomment-266623484

Wondering why we don't default to opaque when user gives `const Color(0x123456)`.

**Comment by Hixie** [comment]
There's no difference between 0x123456 and 0x00123456.

**Comment by xster** [comment]
That was my question. Why don't we check if argument & 0xFF000000 is 0 and add 0xFF000000?

**Comment by abarth** [comment]
@xster Seems like added magic that would be more confusing.  Maybe we should create a new constructor that requires the high bits to be zero?

```
Color c = const Color.rgb(0x123456)
```

Would be opaque rather than transparent.

**Comment by Hixie** [comment]
@xster how would you express "transparent green" then? (e.g. if you wanted to fade something in from transparent to opaque green, but without fading the color at the same time)

@abarth it wouldn't solve the problem, which is people forgetting to give the alpha value when using the `new Color` constructor. If they remembered that that was an issue enough to use `new Color.rgb`, they wouldn't need it in the first place.

**Comment by abarth** [comment]
@Hixie If you're willing to break compat, you can change the current default constructor to be named `Color.argb` to remind callers of the `a` component.

**Comment by Hixie** [comment]
That's a non-starter, it would break almost every app.

**Comment by abarth** [comment]
Another idea: teach the analyzer to complain if you pass a hex literal to the `Color` constructor with less than 8 digits.  The analyzer could even suggest filling the last two digits with FF.  This would require folks who want transparent to be explicit and include leading zeros in their hex literals.

**Comment by Hixie** [comment]
Yeah, a lint is definitely the way to go IMHO. See https://github.com/flutter/flutter/issues/1220 for that and other ideas of Flutter lints we want to eventually add.

**Comment by xster** [comment]
A lint sgtm

**Pull request merged**